### ISSUE TITLE
`kitty @ set-colors` issue

### ISSUE THREAD

**Issue by SolitudeSF** [kovidgoyal/kitty#632]
after running `kitty @ set-colors -a "$XDG_CONFIG_HOME/kitty/colors.conf"` to reapply default colors some colors get changed to incorrect values, but *only* in the terminal its ran in. it works only with `-a` flag and it only works in one window from `--single-instance` group. further invocations in other windows of the same group have no effect.

![2018-06-12_22 38 29](https://user-images.githubusercontent.com/10168645/41313190-fc6d1582-6e91-11e8-82d6-0839cc27f822.png)
![2018-06-12_22 38 49](https://user-images.githubusercontent.com/10168645/41313198-01317c20-6e92-11e8-9b4d-189900a9abd2.png)

**Comment by kovidgoyal** [comment]
Changing colors is supposed to change them only in one window, unless you use the --all and --configured options. And running it with kitty.conf will not change all colors, since kitty.conf typically does not contain all color definitions, only the first 16 out of 256

**Comment by SolitudeSF** [comment]
you misunderstood. i do provide `--all` flag. the problem is that in the window the command is and *only* in that window run some colors get changed to wrong ones.

**Comment by kovidgoyal** [comment]
Hmm, I cannot replicate that, can you provide a minimal example? Something like

kitty @ set-colors foreground=red background=gray

or whatever to recreate the issue.

**Comment by SolitudeSF** [comment]
i couldnt recreate it by specifying colors on the commandline.
but i can reliably recrate it by sourcing this `colors.conf`:
```
color0  #4a3637
color8  #402e2e
color1  #d17b49
color9  #ac5d2f
color2  #7b8748
color10 #647035
color3  #af865a
color11 #8f6840
color4  #535c5c
color12 #444b4b
color5  #775759
color13 #614445
color6  #6d715e
color14 #585c49
color7  #c0b18b
color15 #978965
```
with `kitty @ set-colors -a colors.conf`. also, in my experience it affects the colors that is supposed to be red.

**Comment by SolitudeSF** [comment]
actually, you dont need to specify all colors. the red color in the target window gets changed to whatever color was specified last in the config.

**Comment by SolitudeSF** [comment]
thank you

**Comment by LnLcFlx** [comment]
I still have this issue in 0.17.3.

**Comment by knezi** [comment]
> I still have this issue in 0.17.3.

First, this is fixed in 0.17.4, not 0.17.**3**.

Second, make sure to distinguish between *kitty window* and *wayland window*. Remote control only affects windows within one kitty instance. This is quite confusing and maybe should be better specified in the documentation.
